lib/utils/deploy-legacy.js

```diff
@@ -0,0 +1,194 @@
+	const streamUtils = require('./streams');
+
+	const image = docker.getImage(imageId);
+	const imageMetadata = image.inspect();
+
+	return Promise.join(
+		image.get(),
+		imageMetadata.get('Size'),
+		(imageStream, imageSize) =>
+			streamUtils.buffer(imageStream, bufferFile).tap(bufferedStream => {
+				// @ts-ignore adding an extra property
+				bufferedStream.length = imageSize;
+			}),
+	);
+};
+
+const showPushProgress = function(message) {
+	const visuals = getVisuals();
+	const progressBar = new visuals.Progress(message);
+	progressBar.update({ percentage: 0 });
+	return progressBar;
+};
+
+const uploadToPromise = (uploadRequest, logger) =>
+	new Promise(function(resolve, reject) {
+		const handleMessage = function(data) {
+			let obj;
+			data = data.toString();
+			logger.logDebug(`Received data: ${data}`);
+
+			try {
+				obj = JSON.parse(data);
+			} catch (e) {
+				logger.logError('Error parsing reply from remote side');
+				reject(e);
+				return;
+			}
+
+			switch (obj.type) {
+				case 'error':
+					reject(new Error(`Remote error: ${obj.error}`));
+					break;
+				case 'success':
+					resolve(obj);
+					break;
+				case 'status':
+					logger.logInfo(obj.message);
+					break;
+				default:
+					reject(
+						new Error(`Received unexpected reply from remote: ${data}`),
+					);
+			}
+		};
+
+		uploadRequest.on('error', reject).on('data', handleMessage);
+	});
+
+const uploadImage = function(
+	imageStream,
+	token,
+	username,
+	url,
+	appName,
+	logger,
+) {
+	const request = require('request');
+	const progressStream = require('progress-stream');
+	const zlib = require('zlib');
+
+	// Need to strip off the newline
+	const progressMessage = logger
+		.formatMessage('info', 'Uploading')
+		.slice(0, -1);
+	const progressBar = showPushProgress(progressMessage);
+	const streamWithProgress = imageStream.pipe(
+		progressStream(
+			{
+				time: 500,
+				length: imageStream.length,
+			},
+			({ percentage, eta }) =>
+				progressBar.update({
+					percentage: Math.min(percentage, 100),
+					eta,
+				}),
+		),
+	);
+
+	const uploadRequest = request.post({
+		url: getBuilderPushEndpoint(url, username, appName),
+		headers: {
+			'Content-Encoding': 'gzip',
+		},
+		auth: {
+			bearer: token,
+		},
+		body: streamWithProgress.pipe(
+			zlib.createGzip({
+				level: 6,
+			}),
+		),
+	});
+
+	return uploadToPromise(uploadRequest, logger);
+};
+
+const uploadLogs = function(logs, token, url, buildId, username, appName) {
+	const request = require('request');
+	return request.post({
+		json: true,
+		url: getBuilderLogPushEndpoint(url, buildId, username, appName),
+		auth: {
+			bearer: token,
+		},
+		body: Buffer.from(logs),
+	});
+};
+
+/*
+opts must be a hash with the following keys:
+
+- appName: the name of the app to deploy to
+- imageName: the name of the image to deploy
+- buildLogs: a string with build output
+- shouldUploadLogs
+*/
+export const deployLegacy = function(
```

Comment:
Changed this to `{ deployLegacy }` because `export =` isn't available in js files